### ISSUE TITLE
[a11y]: Notification kind type is not conveyed by screen readers

### ISSUE THREAD

**Issue by guidari** [carbon-design-system/carbon#16391]
### Package

@carbon/react

### Browser

Firefox

### Package version

v1.56.0

### React version

18.02

### Automated testing tool and ruleset

Manual QA

### Assistive technology

JAWS

### Description

The type of notification is not conveyed by the screen reader. Change the `kind` prop in the playground

### Recommended Fix
Ensure that type of notification is conveyed by the screen reader as well

### Reproduction/example

https://react.carbondesignsystem.com/?path=/story/components-notifications-actionable--playground

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [X] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems

**Comment by tay1orjones** [comment]
The icon description [is enough](https://ibm-studios.slack.com/archives/G01GCBCGTPV/p1723040183579489?thread_ts=1722976173.510009&cid=G01GCBCGTPV) to convey the type of notification, provided that description is accurate. So this can be fixed with the following:

- [ ] Set `aria-hidden="false"` on the icon in NotificationIcon to override the default coming from the icon components.
- [ ] Add a default parameter for NotificationIcon's `iconDescription`. [Earlier parameters can be referred to in the initializers of later parameters](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/Default_parameters#:~:text=earlier%20parameters%20can%20be%20referred%20to%20in%20the%20initializers%20of%20later%20parameters) so ``iconDescription = `${kind} notification icon` `` should be possible
- [ ] Modify the `` iconDescription={statusIconDescription || `${kind} icon`} `` usages from all 4 notification components to just `iconDescription={statusIconDescription}`
- [ ] Remove any `iconDescription` value/default overrides used in storybook to ensure storybook always shows the new default string

**Comment by heloiselui** [comment]
Since this playground example doesn’t exist in our Storybook (@carbon/react v1.84), we’re not able to reproduce or verify the issue as described. I believe we can close this issue.
